### ISSUE TITLE
Scrolling on carousel causing default fallback intent to trigger

### ISSUE THREAD

**Issue by benl22** [jovotech/jovo-framework#605]
## I'm submitting a...

- [x] Bug report
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:

## Expected Behavior

My voice app uses a google assistant carousel for displaying a number of options (based on code from here: https://www.jovo.tech/docs/google-assistant/visual-output#carousel-selector). I expect to be able to scroll through the list of options in the carousel and tap to select one.

## Current Behavior

It works as intended on a phone but when attempting to scroll through the options on a Google Home Hub, the default fallback intent is triggered as soon as I start to scroll, so instead of being able to scroll - I am sent to the default fallback screen in my app. This happens if I attempt to scroll before the ask dialogue has finished speaking. If I wait until the microphone opens and then scroll the issue doesn't occur. Is there anything else I would need to implement to enable scrolling?

This is the request I receive when starting to scroll the screen:

`
{
	"responseId": "1d06e2a7-d254-4be6-8054-22fdd84576c7-d5ae01f3",
	"queryResult": {
		"parameters": {},
		"allRequiredParamsPresent": true,
		"outputContexts": [
			{
				"name": "projects/my-project-ksy/agent/sessions/ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0/contexts/actions_capability_media_response_audio"
			},
			{
				"name": "projects/my-project-ksy/agent/sessions/ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0/contexts/actions_capability_account_linking"
			},
			{
				"name": "projects/my-project-ksy/agent/sessions/ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0/contexts/actions_capability_screen_output"
			},
			{
				"name": "projects/my-project-ksy/agent/sessions/ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0/contexts/actions_capability_audio_output"
			},
			{
				"name": "projects/my-project-ksy/agent/sessions/ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0/contexts/google_assistant_input_type_voice"
			}
		],
		"intent": {
			"name": "projects/my-project-ksy/agent/intents/0fcf79b8-1a6b-4501-98a9-b5486bf15bb5",
			"displayName": "Default Fallback Intent",
			"isFallback": true
		},
		"intentDetectionConfidence": 1,
		"languageCode": "en"
	},
	"originalDetectIntentRequest": {
		"source": "google",
		"version": "2",
		"payload": {
			"user": {
				"locale": "en-GB",
				"lastSeen": "2019-10-28T09:59:11Z",
				"userStorage": "{\"userId\":\"d29d842a-5d03-4a9a-b2da-0376d2820f2f\"}",
				"userVerificationStatus": "VERIFIED"
			},
			"conversation": {
				"conversationId": "ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0",
				"type": "ACTIVE",
				"conversationToken": "[]"
			},
			"inputs": [
				{
					"intent": "actions.intent.TEXT",
					"rawInputs": [
						{
							"inputType": "VOICE"
						}
					],
					"arguments": [
						{
							"name": "text"
						}
					]
				}
			],
			"surface": {
				"capabilities": [
					{
						"name": "actions.capability.MEDIA_RESPONSE_AUDIO"
					},
					{
						"name": "actions.capability.ACCOUNT_LINKING"
					},
					{
						"name": "actions.capability.SCREEN_OUTPUT"
					},
					{
						"name": "actions.capability.AUDIO_OUTPUT"
					}
				]
			},
			"isInSandbox": true,
			"availableSurfaces": [
				{
					"capabilities": [
						{
							"name": "actions.capability.SCREEN_OUTPUT"
						},
						{
							"name": "actions.capability.WEB_BROWSER"
						},
						{
							"name": "actions.capability.AUDIO_OUTPUT"
						}
					]
				}
			]
		}
	},
	"session": "projects/my-project-ksy/agent/sessions/ABwppHH9bqu3KYaf59z6DpqiTXpSuAhPQfSYuHsQ8OFOI_Hi1DhWT7D-fsmukn8eyhaXDuI5la8z3pjLQAT3NGrxvk1QxetNulHbVF0"
}
`

### Error log
```
If you have an error log, please paste it here.
```

## Your Environment

* Jovo Framework version used: 2.2.19
* Operating System: macOS 10.14.4
* Running locally using Jovo run - but also same issue when running on Lambda

**Comment by aswetlow** [comment]
It has been delivered :)

I can't reproduce the issue. Can you share some code?

This is what I used:

```javascript
 let carousel = new Carousel();

        carousel.addItem(
            (new OptionItem())
                .setTitle('Text 1')
                .setDescription('Desc 1')
                .setImage({
                    url: 'http://via.placeholder.com/450x350?text=Text1',
                    accessibilityText: 'accessibilityText'})
                .setKey('Carouselitem1key1')
        );
        carousel.addItem(
            (new OptionItem())
                .setTitle('Text 2')
                .setDescription('Desc 2')
                .setImage({
                    url: 'http://via.placeholder.com/450x350?text=Text2',
                    accessibilityText: 'accessibilityText'})
                .setKey('Carouselitem1key2')
        );
        carousel.addItem(
            (new OptionItem())
                .setTitle('Text 3')
                .setDescription('Desc 2')
                .setImage({
                    url: 'http://via.placeholder.com/450x350?text=Text3',
                    accessibilityText: 'accessibilityText'})
                .setKey('Carouselitem1key3')
        );
        this.$googleAction.showCarousel(carousel);

        this.ask('Choose from list', 'Choose from list');
```

**Comment by benl22** [comment]
Thanks for looking into this. I used the code you provided above but I still got the same issue.

This is the response I get when asking for the intent:

`{
        "fulfillmentText": "Choose from list Choose from list Choose from list Choose from list Choose from list",
        "outputContexts": [],
        "payload": {
                "google": {
                        "expectUserResponse": true,
                        "richResponse": {
                                "items": [
                                        {
                                                "simpleResponse": {
                                                        "ssml": "<speak>Choose from list Choose from list Choose from list Choose from list Choose from list</speak>"
                                                }
                                        }
                                ]
                        },
                        "noInputPrompts": [
                                {
                                        "ssml": "<speak>Choose from list</speak>"
                                }
                        ],
                        "systemIntent": {
                                "intent": "actions.intent.OPTION",
                                "data": {
                                        "@type": "type.googleapis.com/google.actions.v2.OptionValueSpec",
                                        "carouselSelect": {
                                                "items": [
                                                        {
                                                                "optionInfo": {
                                                                        "key": "Carouselitem1key1",
                                                                        "synonyms": []
                                                                },
                                                                "title": "Text 1",
                                                                "description": "Desc 1",
                                                                "image": {
                                                                        "url": "http://via.placeholder.com/450x350?text=Text1",
                                                                        "accessibilityText": "accessibilityText"
                                                                }
                                                        },
                                                        {
                                                                "optionInfo": {
                                                                        "key": "Carouselitem1key2",
                                                                        "synonyms": []
                                                                },
                                                                "title": "Text 2",
                                                                "description": "Desc 2",
                                                                "image": {
                                                                        "url": "http://via.placeholder.com/450x350?text=Text2",
                                                                        "accessibilityText": "accessibilityText"
                                                                }
                                                        },
                                                        {
                                                                "optionInfo": {
                                                                        "key": "Carouselitem1key3",
                                                                        "synonyms": []
                                                                },
                                                                "title": "Text 3",
                                                                "description": "Desc 2",
                                                                "image": {
                                                                        "url": "http://via.placeholder.com/450x350?text=Text3",
                                                                        "accessibilityText": "accessibilityText"
                                                                }
                                                        }
                                                ]
                                        }
                                }
                        },
                        "userStorage": "{\"userId\":\"d29d842a-5d03-4a9a-b2da-0376d2820f2f\"}"
                }
        }
}`

And this is the response I get back when starting to scroll the screen.

`{
        "responseId": "09601f06-59ff-4d16-99a5-c1e506de2c36-36cd004f",
        "queryResult": {
                "parameters": {},
                "allRequiredParamsPresent": true,
                "outputContexts": [
                        {
                                "name": "projects/the-church-of-england-beta-ksy/agent/sessions/ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU/contexts/actions_capability_media_response_audio"
                        },
                        {
                                "name": "projects/the-church-of-england-beta-ksy/agent/sessions/ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU/contexts/actions_capability_audio_output"
                        },
                        {
                                "name": "projects/the-church-of-england-beta-ksy/agent/sessions/ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU/contexts/actions_capability_account_linking"
                        },
                        {
                                "name": "projects/the-church-of-england-beta-ksy/agent/sessions/ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU/contexts/actions_capability_screen_output"
                        },
                        {
                                "name": "projects/the-church-of-england-beta-ksy/agent/sessions/ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU/contexts/google_assistant_input_type_voice"
                        }
                ],
                "intent": {
                        "name": "projects/the-church-of-england-beta-ksy/agent/intents/7e9ba989-56a4-4cde-81cb-5745019dbc7c",
                        "displayName": "Default Fallback Intent",
                        "isFallback": true
                },
                "intentDetectionConfidence": 1,
                "languageCode": "en"
        },
        "originalDetectIntentRequest": {
                "source": "google",
                "version": "2",
                "payload": {
                        "user": {
                                "locale": "en-GB",
                                "lastSeen": "2019-11-04T12:31:19Z",
                                "userStorage": "{\"userId\":\"d29d842a-5d03-4a9a-b2da-0376d2820f2f\"}",
                                "userVerificationStatus": "VERIFIED"
                        },
                        "conversation": {
                                "conversationId": "ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU",
                                "type": "ACTIVE",
                                "conversationToken": "[]"
                        },
                        "inputs": [
                                {
                                        "intent": "actions.intent.TEXT",
                                        "rawInputs": [
                                                {
                                                        "inputType": "VOICE"
                                                }
                                        ],
                                        "arguments": [
                                                {
                                                        "name": "text"
                                                }
                                        ]
                                }
                        ],
                        "surface": {
                                "capabilities": [
                                        {
                                                "name": "actions.capability.MEDIA_RESPONSE_AUDIO"
                                        },
                                        {
                                                "name": "actions.capability.AUDIO_OUTPUT"
                                        },
                                        {
                                                "name": "actions.capability.ACCOUNT_LINKING"
                                        },
                                        {
                                                "name": "actions.capability.SCREEN_OUTPUT"
                                        }
                                ]
                        },
                        "isInSandbox": true,
                        "availableSurfaces": [
                                {
                                        "capabilities": [
                                                {
                                                        "name": "actions.capability.AUDIO_OUTPUT"
                                                },
                                                {
                                                        "name": "actions.capability.SCREEN_OUTPUT"
                                                },
                                                {
                                                        "name": "actions.capability.WEB_BROWSER"
                                                }
                                        ]
                                }
                        ]
                }
        },
        "session": "projects/the-church-of-england-beta-ksy/agent/sessions/ABwppHG7tTKnqpQj05hZfywHYImuBZMf7qd2qbc8TGo3BDxwT9NjMiOeJFZcoR4jyhmP-87MOcNnqeQncnC0rQ1UvupSH1whatZ1XjU"
}`

The only thing I changed from your code is giving the ask a longer response string to talk longer (as the issue only occurs when scrolling whilst the assistant is still speaking).

When I start scrolling the screen - I receive the 'ON_REQUEST' response (where I currently don't do anything).

**Comment by aswetlow** [comment]
Oh ok, now I can reproduce it. I don't have a solution yet and will keep you posted.

**Comment by benl22** [comment]
Hi - is there any update on this? It's fine if not - I'm about to release an update but will hold off if you've got any ideas. Thanks for your help and this great Library!

**Comment by alexreiling** [comment]
Same issue here. Also on a Google Home Hub. Will this be fixed? Thanks...

**Comment by vc-ca** [comment]
@benl22 Thanks so much for submitting this! I've been banging my head against the wall trying to figure this out. Glad it's not just me 🤣

**Comment by marcd123** [comment]
I'm also experiencing this issue with scrolling Carousel's and List's. Is there any news on a solution?

**Comment by aswetlow** [comment]
I can't reproduce it anymore. Is there a fix that hasn't been rolled out for everyone?

The Google Assistant stops talking after the first scroll, but it doesn't send a request.

@marcd123  @vc-ca @alexreiling @benl22 
What's your device's firmware version? Mine is 1.44.191160


Here's the code:
```
let carousel = new Carousel();

      carousel.addItem(
        (new OptionItem())
          .setTitle('Text 1')
          .setDescription('Desc 1')
          .setImage({
            url: 'http://via.placeholder.com/450x350?text=Text1',
            accessibilityText: 'accessibilityText'})
          .setKey('Carouselitem1key1')
      );
      carousel.addItem(
        (new OptionItem())
          .setTitle('Text 2')
          .setDescription('Desc 2')
          .setImage({
            url: 'http://via.placeholder.com/450x350?text=Text2',
            accessibilityText: 'accessibilityText'})
          .setKey('Carouselitem1key2')
      );
      carousel.addItem(
        (new OptionItem())
          .setTitle('Text 3')
          .setDescription('Desc 2')
          .setImage({
            url: 'http://via.placeholder.com/450x350?text=Text3',
            accessibilityText: 'accessibilityText'})
          .setKey('Carouselitem1key3')
      );
      this.$googleAction.showCarousel(carousel);

      this.ask('Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.', 'Choose from list');
```

**Comment by vc-ca** [comment]
The problem seems to be related to the Google Nest Hub firmware version.

I was able to resolve this issue by joining Googles "[Preview Program](https://support.google.com/googlenest/answer/6343937?hl=en)", which updated my Nest Hub firmware.

Previous firmware version: 1.42.171872  
Current firmware :  1.44.191160

I submitted a trouble ticket on [https://developers.google.com/assistant](https://developers.google.com/assistant) a month ago, and just today heard: 

"Kindly note that we are already tracking the issue internally and plan to address it."

Billion dollar companies don't move quick, huh?
